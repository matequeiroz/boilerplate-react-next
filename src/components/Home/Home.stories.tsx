import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Home } from './Home'

export default {
  title: 'Home',
  component: Home
} as ComponentMeta<typeof Home>

const Template: ComponentStory<typeof Home> = (args) => <Home {...args} />

export const FirstStory = Template.bind({})

FirstStory.args = {
  title: 'React Avan√ßado',
  description: 'React, Next, Styled-Components e TypeScript'
}
