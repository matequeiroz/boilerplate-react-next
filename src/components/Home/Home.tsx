import * as S from './styles'

export const Home: React.FC = () => (
  <S.Wrapper>
    <S.Logo
      src={'/img/logo.svg'}
      alt="Um átomo de cor branca com fundo rosa e a frase 'React Avançado'"
      data-testid="home-logo"
    />
    <S.Title data-testid="home-title">React Avançado</S.Title>
    <S.Description data-testid="home-description">
      ReactJS, Typescript, NextJS e Styled-Components
    </S.Description>
    <S.Illustration
      data-testid="home-illustration"
      src={'/img/hero-illustration.svg'}
      alt="Figura de desenvolvedor sentado com telas com código a sua frente"
    />
  </S.Wrapper>
)
