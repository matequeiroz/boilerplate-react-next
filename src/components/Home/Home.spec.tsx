import { render, screen } from '@testing-library/react'

import { Home } from './Home'

describe('Home component', () => {
  it('should render home component correctly', () => {
    const { container } = render(<Home />)
    expect(screen.getByTestId('home-logo')).toBeInTheDocument()
    expect(screen.getByTestId('home-title')).toBeInTheDocument()
    expect(screen.getByTestId('home-description')).toBeInTheDocument()
    expect(screen.getByTestId('home-illustration')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
