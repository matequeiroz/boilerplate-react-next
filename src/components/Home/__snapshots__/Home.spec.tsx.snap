// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render home component correctly 1`] = `
<main
  class="styles__Wrapper-sc-117wb7c-0 dBBhzO"
>
  <img
    alt="Um átomo de cor branca com fundo rosa e a frase 'React Avançado'"
    class="styles__Logo-sc-117wb7c-1 dRFvdt"
    data-testid="home-logo"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-117wb7c-3 eIeHoi"
    data-testid="home-title"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-sc-117wb7c-4 cYOXqX"
    data-testid="home-description"
  >
    ReactJS, Typescript, NextJS e Styled-Components
  </h2>
  <img
    alt="Figura de desenvolvedor sentado com telas com código a sua frente"
    class="styles__Illustration-sc-117wb7c-2 kUbtBf"
    data-testid="home-illustration"
    src="/img/hero-illustration.svg"
  />
</main>
`;
